### ISSUE TITLE
Support for compilation / Makefile execution

### ISSUE THREAD

**Issue by p0pr0ck5** [openresty/opm#30]
Any plans to support OPM packages to execute `install` commands when installed via opm? E.g. a package could execute a command to locally make some C module as part of the installation process, similar to what Luarocks offers.

**Comment by agentzh** [comment]
@p0pr0ck5 Support for Lua libraries with C components are already planned. See our TODO list:

https://github.com/openresty/opm#todo

But we won't use the author Makefile since Makefiles are non-portable per se. We will provide our own mechanism in opm.

**Comment by isage** [comment]
>we won't use the author Makefile since Makefiles are non-portable per se. We will provide our own mechanism in opm.

I, for example, use CMake. And CMake is portable. As an author, it would be a burden to me to maintain some different (opm) build system configuration alongside cmake one.

**Comment by agentzh** [comment]
@isage cmake is an extra dependency that I never like.

**Comment by isage** [comment]
Yes, but let's take LuaRocks, for example: i can use any build system i want. Makefiles, CMake or just describe build steps in rockspec. That way i can reuse my already existing build scenario.
If external build systems won't be supported i'd rather not support opm and, provided opm supports LuaRocks at some point, support just LuaRocks.

**Comment by agentzh** [comment]
@isage The use of LuaRocks is deprecated in OpenResty. opm decides to do its own build system for C components, in a very portable way.

**Comment by isage** [comment]
I've meant luarocks/ namespace support, as mentioned in TODO.
As for portability, well, opm can't be used outside of openresty, can't be used for creating deb/rpm/etc., so.
Point is, i, as a module author, wouldn't want to maintain both CMake/Makefile/Whatever build config and opm one. Or ditch first in favor of opm.

**Comment by agentzh** [comment]
@isage Well, the LuaRocks libraries' config will get converted to opm's build system, at least for simple ones. That's how the compatibility will work out. Opm will never run LuaRock module's own build systems directly on  opm clients' machines.
